<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SingCellaR</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SingCellaR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="Introduction.html">Introduction</a>
</li>
<li>
  <a href="SingCellaR_workflow1.html">Standard analysis workflow for PBMCs</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Integrative analysis from multiple samples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">Individual sample analysis</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample1.html">Healthy donor sample 1</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample6.html">Healthy donor sample 6</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample11.html">Myelofibrosis patient 11</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample20.html">Myelofibrosis patient 20</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample21.html">Myelofibrosis patient 21</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="SingCellaR_workflow_Psaila_et_al_integration.html">Integrative analysis from multiple samples</a>
    </li>
    <li>
      <a href="SingCellaR_workflow_Psaila_et_al_integration_analysis.html">Integrative downstream analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="SingCellaR_workflow_TARGET_Seq.html">TARGET-Seq analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="mailto:supat.thongjuea@imm.ox.ac.uk">
    <span class="fa fa-envelope"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">SingCellaR</h1>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"images/SingCellaR.png\" style=\"position:static; top:0; right:0; padding:20px; height:180px\"></div>');
  $('#header').css('margin-right', '180px')
});
</script>
<div id="section" class="section level2">
<h2></h2>
<p>The SingCellaR package provides a comprehensive set of R functions necessary for the analysis of single-cell RNA-seq data. The package includes facilities for:</p>
<ul>
<li>
supporting the analysis for different types of single-cell technologies (e.g. Chromium 10x genomics, Smart-Seq2, and drop-let based technologies
</li>
<li>
<p>supporting Target-Seq analysis which incorporates both single-cell mutational and single-cell transcriptome profiling in the same single cell.</p>
<li>
Data filtering and normalization
</li>
<li>
Confounding factors removal by regressing out unwanted source of variations. For Target-Seq, it supports analysis to preserve known genotypes to prevent the removal of biological meaning.
</li>
<li>
<p>Data integration, SingCellaR incorporates different types of existing integrative methods:</p>
<pre><code>    Harmony,
    Seurat version3,
    Liger,
    Scanorama,
    Limma,
    and Combat. </code></pre>
It facilitates downstream analysis for each type of data integration within the same object. So, the users can easily visualise the results such as UMAP, tSNE, and etc. directly without generation the R object for each separated integration method.
</li>
<li>
Supporting our developed integration method named “supervised harmony”. This method is used as an additional algorithm to integrate data which have similar cell-types (e.g. CD34+Lin-) generated from different batches and environmental conditions.
</li>
<li>
Cluster identification, it supports clustering analyses both graph-based community detection and k-means clustering.
</li>
<li>
Cell-type annotation, we developed the semi-automate cell-type annotation system that can help to assign the cell types to identified clusters. This system produces the enrichment scores of possible cell types derived from multiple curated signature gene sets (the users can input their custom made of signature genes), which can be greatly help to assign the cell types.
</li>
<li>
Cell trajectory analysis, SingCellaR incorporates cell trajectory analysis based on force-directed graph, KNN-graph, and diffusion map and develop the gene set scoring system that can superimpose on top of the embeddings facilitating the cell type and state annotation.
</li>
<pre><code>   Force-directed graph
   KNN-Graph
   Diffusion Map</code></pre>
<li>
Differential gene expression, SingCellaR combines nonparametric Wilcoxon test on the log-transformed, normalized UMIs to compare expression level and Fisher′s exact test that uses to compare the cell frequency expressing each gene as previously described (Giustacchini et al., Nature Medicine, 2017). P values generated from both tests are then combined using Fisher’s method and are adjusted using the Benjamini-Hochberg (BH) correction.
</li>
<li>
Gene set enrichment analysis, SingCellaR incorporates fGSEA <a href="https://bioconductor.org/packages/release/bioc/html/fgsea.html" class="uri">https://bioconductor.org/packages/release/bioc/html/fgsea.html</a> for GSEA analysis
</li>
<li>
Powerful visualization to deliver interpretable biological meaning
</li>
<pre><code>   tSNE
   UMAP
   Force-directed graph
   KNN-graph visualisation in both 2D and 3D
   Diffusion map
   Violin plot
   Bubble plot
   Heatmap
   
   Different types of plots can be visualised with different types of information :
   - Individual gene expression
   - Gene set score
   - AUCell score
   </code></pre>
</ul>
</div>
<div id="getting-started" class="section level2">
<h2>Getting started</h2>
<p>Installation</p>
<pre><code>install.packages(&#39;devtools&#39;)
library(devtools)
install_github(&#39;supatt-lab/SingCellaR&#39;)</code></pre>
</div>
<div id="simple-method-to-install-python-packages-in-r-environment" class="section level2">
<h2>Simple method to install python packages in R environment</h2>
<p>Here’s an example of using these functions to create an environment, install packages within it, then use the environment from R:</p>
<pre><code>library(reticulate)
#create a new environment 
conda_create(&quot;r-reticulate&quot;)</code></pre>
<p>The reticulate will create the r-miniconda in your working environment (e.g. in /Users/supat/Library/r-miniconda/envs)</p>
<p>To install python library use:</p>
<pre><code>py_install(&quot;fa2&quot;)</code></pre>
<p>More information can be found from <a href="https://rstudio.github.io/reticulate/articles/python_packages.html#conda-installation-1" class="uri">https://rstudio.github.io/reticulate/articles/python_packages.html#conda-installation-1</a></p>
<p>To install Scanorama, the standard ‘pip’ software found in ‘r-miniconda environment’ can be used :</p>
<pre><code>./pip install scanorama</code></pre>
</div>
<div id="additional-installation-steps-for-macos" class="section level2">
<h2>Additional installation steps for MacOS</h2>
<p>SingCellaR implemented c++ functions that need to be compiled. These functions built on RcppArmadillo, RcppAnnoy, and RcppParallel that require c++ compilation.</p>
<p>For the mac users, the compiler ‘Clang’ has to be installed <a href="https://clang.llvm.org/get_started.html" class="uri">https://clang.llvm.org/get_started.html</a></p>
<p>The developer uses Clang6 and it works fine for the macOs Catilina with R3.6.</p>
<p>Install gfortran as suggested here <a href="https://gcc.gnu.org/wiki/GFortranBinaries" class="uri">https://gcc.gnu.org/wiki/GFortranBinaries</a></p>
<p>Install Clang as suggested here <a href="https://clang.llvm.org/get_started.html" class="uri">https://clang.llvm.org/get_started.html</a></p>
<p>The easy way is to download the pre-built binaries and uncompress the resulting zip file and type into terminal (sudo if needed):</p>
<pre><code>mv /path/to/clang6/ /usr/local/ 

Create .R/Makevars file containing following:

# The following statements are required to use the clang6 binary
CC=/usr/local/clang6/bin/clang
CXX=/usr/local/clang6/bin/clang++
CXX11=/usr/local/clang6/bin/clang++
CXX14=/usr/local/clang6/bin/clang++
CXX17=/usr/local/clang6/bin/clang++
CXX1X=/usr/local/clang6/bin/clang++
LDFLAGS=-L/usr/local/clang6/lib</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
