<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>SingCellaR analysis workflow for 10x Genomics scRNA-seq data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cerulean.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">SingCellaR</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="Introduction.html">Introduction</a>
</li>
<li>
  <a href="SingCellaR_workflow1.html">Standard analysis workflow for PBMCs</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Integrative analysis from multiple samples
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li class="dropdown-submenu">
      <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">Individual sample analysis</a>
      <ul class="dropdown-menu" role="menu">
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample1.html">Healthy donor sample 1</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample6.html">Healthy donor sample 6</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample11.html">Myelofibrosis patient 11</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample20.html">Myelofibrosis patient 20</a>
        </li>
        <li>
          <a href="SingCellaR_workflow_Psaila_et_al_sample21.html">Myelofibrosis patient 21</a>
        </li>
      </ul>
    </li>
    <li>
      <a href="SingCellaR_workflow_Psaila_et_al_integration.html">Integrative analysis from multiple samples</a>
    </li>
    <li>
      <a href="SingCellaR_workflow_Psaila_et_al_integration_analysis.html">Integrative downstream analyses</a>
    </li>
  </ul>
</li>
<li>
  <a href="SingCellaR_workflow_TARGET_Seq.html">TARGET-Seq analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="mailto:supat.thongjuea@imm.ox.ac.uk">
    <span class="fa fa-envelope"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">SingCellaR analysis workflow for 10x
Genomics scRNA-seq data</h1>

</div>


<script type="text/javascript">
$(document).ready(function() {
  $('#header').parent().prepend('<div id=\"logo\"><img src=\"images/SingCellaR.png\" style=\"position:static; top:0; right:0; padding:20px; height:180px\"></div>');
  $('#header').css('margin-right', '180px')
});
</script>
<div
id="standard-analysis-workflow-for-a-healthy-donor-sample1-generated-by-psaila-et-al.-mol.-cell-2020"
class="section level2">
<h2>Standard analysis workflow for a healthy donor (Sample1) generated
by Psaila et al., Mol. Cell, 2020</h2>
<pre class="r"><code>library(SingCellaR)

data_matrices_dir&lt;-&quot;../SingCellaR_example_datasets/Psaila_et_al/Sample01_cellranger/&quot;
Sample1&lt;-new(&quot;SingCellaR&quot;)
Sample1@dir_path_10x_matrix&lt;-data_matrices_dir
Sample1@sample_uniq_id&lt;-&quot;Sample1&quot;

load_matrices_from_cellranger(Sample1,cellranger.version = 3)</code></pre>
<pre><code>## [1] &quot;The sparse matrix is created.&quot;</code></pre>
<pre class="r"><code>Sample1</code></pre>
<pre><code>## An object of class SingCellaR with a matrix of : 33538 genes across 4382 samples.</code></pre>
<p>Cell information will be processed. The percentage of mitochondrial
per cell will be calculated. For human, mitochondrial gene names start
with “MT-”.</p>
<pre class="r"><code>process_cells_annotation(Sample1,mito_genes_start_with =&quot;MT-&quot;)</code></pre>
<pre><code>## [1] &quot;List of mitochondrial genes:&quot;
##  [1] &quot;MT-ND1&quot;  &quot;MT-ND2&quot;  &quot;MT-CO1&quot;  &quot;MT-CO2&quot;  &quot;MT-ATP8&quot; &quot;MT-ATP6&quot; &quot;MT-CO3&quot; 
##  [8] &quot;MT-ND3&quot;  &quot;MT-ND4L&quot; &quot;MT-ND4&quot;  &quot;MT-ND5&quot;  &quot;MT-ND6&quot;  &quot;MT-CYB&quot; 
## [1] &quot;The meta data is processed.&quot;</code></pre>
</div>
<div id="qc-plots" class="section level2">
<h2>QC plots</h2>
<h4>
Histogram
</h4>
<pre class="r"><code>plot_cells_annotation(Sample1,type=&quot;histogram&quot;)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/cell_QC_histogram-1.png" width="672" /></p>
<h4>
Boxplot
</h4>
<pre class="r"><code>plot_cells_annotation(Sample1,type=&quot;boxplot&quot;)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/cell_QC_boxplot-1.png" width="672" /></p>
<h4>
UMI counts/cell vs. number of detected genes/cell
</h4>
<pre class="r"><code>plot_UMIs_vs_Detected_genes(Sample1)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/cell_genes_vs_UMIs-1.png" width="672" /></p>
</div>
<div id="doublet-detection" class="section level2">
<h2>Doublet detection</h2>
<pre class="r"><code>DoubletDetection_with_scrublet(Sample1)</code></pre>
<pre><code>## [1] &quot;Processing Scrublet..&quot;
## [1] &quot;Doublet prediction result is added in the metadata.&quot;</code></pre>
<pre class="r"><code>table(Sample1@meta.data$Scrublet_type)</code></pre>
<pre><code>## 
## Doublets  Singlet 
##        5     4377</code></pre>
<p>From the plots above, we set up the cutoff criteria as described in
the followings :</p>
<ul>
<li>
min_UMIs=1000
</li>
<li>
max_UMIs=60000
</li>
<li>
min_detected_genes=500
</li>
<li>
max_detected_genes=6000
</li>
<li>
max_percent_mito=15
</li>
<li>
genes_with_expressing_cells=10
</li>
</ul>
</div>
<div id="filtering" class="section level2">
<h2>Filtering</h2>
<pre class="r"><code>filter_cells_and_genes(Sample1,
                       min_UMIs=1000,
                       max_UMIs=60000,
                       min_detected_genes=500,
                       max_detected_genes=6000,
                       max_percent_mito=15,
                       genes_with_expressing_cells = 10,isRemovedDoublets = T)</code></pre>
<pre><code>## [1] &quot;The cells and genes metadata are updated by adding the filtering status.&quot;
## [1] &quot;1113/4382 cells will be filtered out from the downstream analyses!.&quot;</code></pre>
</div>
<div id="normalization" class="section level2">
<h2>Normalization</h2>
<pre class="r"><code>normalize_UMIs(Sample1,use.scaled.factor = T)</code></pre>
<pre><code>## [1] &quot;Normalization is completed!.&quot;</code></pre>
</div>
<div id="regressing-unwanted-source-of-variations"
class="section level2">
<h2>Regressing unwanted source of variations</h2>
<p>Next, the effect of the library size and percentage of mitochondrial
reads are regressed out.</p>
<pre class="r"><code>remove_unwanted_confounders(Sample1,residualModelFormulaStr=&quot;~UMI_count+percent_mito&quot;)</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |====                                                                  |   6%
  |                                                                            
  |========                                                              |  12%
  |                                                                            
  |============                                                          |  18%
  |                                                                            
  |================                                                      |  24%
  |                                                                            
  |=====================                                                 |  29%
  |                                                                            
  |=========================                                             |  35%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |=================================                                     |  47%
  |                                                                            
  |=====================================                                 |  53%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |=============================================                         |  65%
  |                                                                            
  |=================================================                     |  71%
  |                                                                            
  |======================================================                |  76%
  |                                                                            
  |==========================================================            |  82%
  |                                                                            
  |==============================================================        |  88%
  |                                                                            
  |==================================================================    |  94%
  |                                                                            
  |======================================================================| 100%
## [1] &quot;Removing unwanted sources of variation is done!&quot;</code></pre>
</div>
<div id="variable-gene-detection" class="section level2">
<h2>Variable gene detection</h2>
<pre class="r"><code>get_variable_genes_by_fitting_GLM_model(Sample1,mean_expr_cutoff = 0.05,disp_zscore_cutoff = 0.05)</code></pre>
<pre><code>## [1] &quot;Calculate row variance..&quot;
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |================                                                      |  22%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |======================================================                |  78%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |======================================================================| 100%[1] &quot;Using :12749 genes for fitting the GLM model!&quot;
## [1] &quot;Identified :2421 variable genes&quot;</code></pre>
<p>House keeping (e.g., ribosomal genes) and mitochondrial genes should
be removed from the list of variable genes. SingCellaR reads in the GMT
file that contains ribosomal and mitochondrial genes and removes these
genes from the list of highly variable genes. Below shows the example
for removing genes.</p>
<pre class="r"><code>remove_unwanted_genes_from_variable_gene_set(Sample1,gmt.file = &quot;../SingCellaR_example_datasets/Human_genesets/human.ribosomal-mitocondrial.genes.gmt&quot;,
                                             removed_gene_sets=c(&quot;Ribosomal_gene&quot;,&quot;Mitocondrial_gene&quot;))</code></pre>
<pre><code>## [1] &quot;3 genes are removed from the variable gene set.&quot;</code></pre>
<p>Here, the plot shows highly variable genes (cyan) in the fitted GLM
model.</p>
<pre class="r"><code>plot_variable_genes(Sample1)</code></pre>
<pre><code>## [1] &quot;Calculate row variance..&quot;
## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |================                                                      |  22%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |======================================================                |  78%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |======================================================================| 100%</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/hvg_plot-1.png" width="672" /></p>
</div>
<div id="pca-anlysis" class="section level2">
<h2>PCA anlysis</h2>
<p>Top 50 PCs will be calculated using IRLBA method from <a
href="https://github.com/bwlewis/irlba"
class="uri">https://github.com/bwlewis/irlba</a> The parameter
“use.regressout.data” is critical. If TRUE, PCA uses the regressed out
data. If FALSE, PCA uses the normalized data without removing unwanted
source of variations.</p>
<pre class="r"><code>runPCA(Sample1,use.components=50,use.regressout.data = T)</code></pre>
<pre><code>## [1] &quot;PCA analysis is done!.&quot;</code></pre>
<p>Next, the number of PCs used for downstream analyses will be
determined from a scree plot.</p>
<pre class="r"><code>plot_PCA_Elbowplot(Sample1)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/plot_PCA_elbow-1.png" width="672" /></p>
</div>
<div id="umap-analysis" class="section level2">
<h2>UMAP analysis</h2>
<pre class="r"><code>runUMAP(Sample1,dim_reduction_method = &quot;pca&quot;,n.dims.use = 20,n.neighbors = 30,
        uwot.metric = &quot;euclidean&quot;)</code></pre>
<pre><code>## 10:46:42 UMAP embedding parameters a = 1.121 b = 1.057</code></pre>
<pre><code>## 10:46:42 Read 3269 rows and found 20 numeric columns</code></pre>
<pre><code>## 10:46:42 Using Annoy for neighbor search, n_neighbors = 30</code></pre>
<pre><code>## 10:46:42 Building Annoy index with metric = euclidean, n_trees = 50</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|
## 10:46:43 Writing NN index file to temp file /var/folders/c6/_w309xr54n7cphf9nw4x_ms40000gn/T//RtmpOsKSli/file13367c83ece4
## 10:46:43 Searching Annoy index using 8 threads, search_k = 3000
## 10:46:43 Annoy recall = 100%
## 10:46:43 Commencing smooth kNN distance calibration using 8 threads
## 10:46:44 Initializing from normalized Laplacian + noise
## 10:46:44 Commencing optimization for 500 epochs, with 138784 positive edges
## 10:46:48 Optimization finished</code></pre>
<pre><code>## [1] &quot;UMAP analysis is done!.&quot;</code></pre>
<pre class="r"><code>plot_umap_label_by_a_feature_of_interest(Sample1,feature = &quot;UMI_count&quot;,point.size = 0.1,mark.feature = FALSE)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/plot_UMAP-1.png" width="672" /></p>
</div>
<div id="cluster-identification" class="section level2">
<h2>Cluster identification</h2>
<p>We further identify clusters using the following function.</p>
<pre class="r"><code>identifyClusters(Sample1,n.dims.use = 30,n.neighbors = 30,knn.metric = &quot;euclidean&quot;)</code></pre>
<pre><code>## [1] &quot;Building Annoy index with metric =  euclidean , n_trees =  50&quot;</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|</code></pre>
<pre><code>## [1] &quot;Searching Annoy index, search_k =  3000&quot;</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%
## [----|----|----|----|----|----|----|----|----|----|
## **************************************************|</code></pre>
<pre><code>## [1] &quot;Building graph network..&quot;
## [1] &quot;Building graph network is done.&quot;
## [1] &quot;Process community detection..&quot;
## [1] &quot;Louvain analysis is done!.&quot;</code></pre>
<p>Identified clusters are then superimposed on top of the UMAP.</p>
<pre class="r"><code>plot_umap_label_by_clusters(Sample1,show_method = &quot;louvain&quot;,point.size = 0.80)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/plot_UMAP_cluster-1.png" width="672" /></p>
</div>
<div id="force-directed-graph-analysis" class="section level2">
<h2>Force-directed graph analysis</h2>
<p>SingCellaR incorporates force-directed graph for the visualization of
cellular trajectories.</p>
<pre class="r"><code>runFA2_ForceDirectedGraph(Sample1,n.dims.use = 20,
                          n.neighbors = 5,n.seed = 1,fa2_n_iter = 1000)</code></pre>
<pre><code>## [1] &quot;Building Annoy index with metric =  euclidean , n_trees =  50&quot;</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%</code></pre>
<pre><code>## [----|----|----|----|----|----|----|----|----|----|</code></pre>
<pre><code>## **************************************************|</code></pre>
<pre><code>## [1] &quot;Searching Annoy index, search_k =  500&quot;</code></pre>
<pre><code>## 0%   10   20   30   40   50   60   70   80   90   100%
## [----|----|----|----|----|----|----|----|----|----|
## **************************************************|</code></pre>
<pre><code>## [1] &quot;Processing fa2..&quot;
## [1] &quot;Force directed graph analysis is done!.&quot;</code></pre>
<p>Force-directed graph is then plotted with identified clusters.</p>
<pre class="r"><code>plot_forceDirectedGraph_label_by_clusters(Sample1,show_method = &quot;louvain&quot;,vertex.size = 0.85,
                                          background.color = &quot;black&quot;)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/plot_FA2_cluster-1.png" width="672" /></p>
</div>
<div id="the-superimposition-of-signature-gene-scores"
class="section level2">
<h2>The superimposition of signature gene scores</h2>
<p>SingCellaR can superimpose signature gene scores from different gene
sets on top of embeddings.</p>
<p>The plots below show the superimposition of gene scores from
erythroid, myeloid, lymphoid, and megakaryocytic gene sets on top UMAP
and force-directed graph respectively.</p>
<pre class="r"><code>plot_umap_label_by_multiple_gene_sets(Sample1,gmt.file = &quot;../SingCellaR_example_datasets/Human_genesets/human.signature.genes.v1.gmt&quot;,
                                      show_gene_sets = c(&quot;Erythroid&quot;,&quot;Lymphoid&quot;,&quot;Myeloid&quot;,&quot;Megakaryocyte&quot;),
                                      custom_color = c(&quot;red&quot;,&quot;orange&quot;,&quot;cyan&quot;,&quot;purple&quot;),
                                      isNormalizedByHouseKeeping = T,point.size = 1,background.color = &quot;black&quot;)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/UMAP_with_genesets-1.png" width="672" /></p>
<pre class="r"><code>plot_forceDirectedGraph_label_by_multiple_gene_sets(Sample1,gmt.file = &quot;../SingCellaR_example_datasets/Human_genesets/human.signature.genes.v1.gmt&quot;,
                                      show_gene_sets = c(&quot;Erythroid&quot;,&quot;Lymphoid&quot;,&quot;Myeloid&quot;,&quot;Megakaryocyte&quot;),
                                      custom_color = c(&quot;red&quot;,&quot;orange&quot;,&quot;cyan&quot;,&quot;purple&quot;),
                                      isNormalizedByHouseKeeping = T,vertex.size = 1,edge.size = 0.1,
                                      background.color = &quot;black&quot;)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/FA2_with_genesets-1.png" width="672" /></p>
</div>
<div id="marker-gene-identification" class="section level2">
<h2>Marker gene identification</h2>
<p>Next, we perform differential gene expression analysis across
clusters to identify marker genes that are highly expressed in each
cluster.</p>
<p><span style="color:red">* This step is critical, prepared for the
“supervised harmony” integration.</span> Supervised harmony requires
differential genes across identified clusters per each sample.</p>
<pre class="r"><code>findMarkerGenes(Sample1,cluster.type = &quot;louvain&quot;)</code></pre>
<pre><code>## [1] &quot;Number of genes for analysis: 16298&quot;
## [1] &quot;Creating binary matrix..&quot;
## [1] &quot;Finding differential genes..&quot;
## [1] &quot;Identifying marker genes for: cl1&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl2&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl3&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl4&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl5&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl6&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl7&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl8&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl9&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl10&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl11&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;
## [1] &quot;Identifying marker genes for: cl12&quot;
## [1] &quot;Processing Fisher&#39;s exact test!&quot;
## [1] &quot;Processing Wilcoxon Rank-sum test!&quot;
## [1] &quot;Combining p-values using the fisher&#39;s method!&quot;</code></pre>
<h4>
The heatmap of top 10 genes
</h4>
<pre class="r"><code>plot_heatmap_for_marker_genes(Sample1,cluster.type = &quot;louvain&quot;,n.TopGenes = 10,rowFont.size = 5,use_raster = FALSE)</code></pre>
<p><img src="SingCellaR_workflow_Psaila_et_al_sample1_files/figure-html/heatmap_marker_genes-1.png" width="960" /></p>
</div>
<div id="save-singcellar-object-for-the-integration"
class="section level2">
<h2>Save SingCellaR object for the integration</h2>
<p>Sample1’s object will be saved for the integration with other
samples.</p>
<pre class="r"><code>save(Sample1,file=&quot;../SingCellaR_example_datasets/Psaila_et_al/SingCellaR_objects/Sample1.SingCellaR.rdata&quot;)</code></pre>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
